vault/token_store.go

```diff
@@ -138,6 +137,77 @@ var (
 )
 
 func (ts *TokenStore) paths() []*framework.Path {
+	commonFieldsForCreate := map[string]*framework.FieldSchema{
+		"display_name": {
+			Type:        framework.TypeString,
+			Description: "Name to associate with this token",
+		},
+		"explicit_max_ttl": {
+			Type:        framework.TypeString,
+			Description: "Explicit Max TTL of this token",
+		},
+		"entity_alias": {
+			Type:        framework.TypeString,
+			Description: "Name of the entity alias to associate with this token",
+		},
+		"num_uses": {
+			Type:        framework.TypeInt,
+			Description: "Max number of uses for this token",
+		},
+		"period": {
+			Type:        framework.TypeString,
+			Description: "Renew period",
+		},
+		"renewable": {
+			Type:        framework.TypeBool,
+			Description: "Allow token to be renewed past its initial TTL up to system/mount maximum TTL",
+			Default:     true,
+		},
+		"ttl": {
+			Type:        framework.TypeString,
+			Description: "Time to live for this token",
+		},
+		"lease": {
+			Type:        framework.TypeString,
+			Description: "Use 'ttl' instead",
+			Deprecated:  true,
+		},
+		"type": {
+			Type:        framework.TypeString,
+			Description: "Token type",
+		},
+		"no_default_policy": {
+			Type:        framework.TypeBool,
+			Description: "Do not include default policy for this token",
+		},
+		"id": {
+			Type:        framework.TypeString,
+			Description: "Value for the token",
+		},
+		"meta": {
+			Type:        framework.TypeKVPairs,
+			Description: "Arbitrary key=value metadata to associate with the token",
+		},
+		"no_parent": {
+			Type:        framework.TypeBool,
+			Description: "Create the token with no parent",
+		},
+		"policies": {
+			Type:        framework.TypeStringSlice,
+			Description: "List of policies for the token",
+		},
+	}
+
+	fieldsForCreateWithRole := map[string]*framework.FieldSchema{
+		"role_name": {
+			Type:        framework.TypeString,
+			Description: "Name of the role",
+		},
+	}
+	for k, v := range commonFieldsForCreate {
+		fieldsForCreateWithRole[k] = v
+	}
```

Comment:
Nit: it might be cleaner to use maps.Copy here but I'm OK leaving it as is since `maps` is still an `x/exp` package.

Reply:
Since `x/exp/maps` has not so far been used anywhere in Vault, and the `x/exp` warning of lack of compatibility is rather strongly worded, I'd prefer not to introduce it to save just a couple of lines of code.